libc/termcap/tputs.c

```diff
@@ -1,11 +1,8 @@
 /* Outputting a string with padding.  */
 
-#ifdef __linux__
-speed_t
-#else
-short
-#endif
-	 ospeed;
+#include <sys/types.h>
+
+speed_t ospeed;
```

Comment:
Hello @ccoffing,

We can't accept this change, as other programs (e.g see elkscmd/elvis/curses.{h,c}) expect `extern short speed`, which seems to be part of the FreeBSD (and macOS) man pages for `tputs(3)`. It seems a bit of a mess, and the Linux man page for tputs has dropped all of this entirely, so perhaps the entire extern declaration should be removed entirely for Linux? (I haven't access for Linux curses.h or term.h to see if this variable exists anymore in Linux).

I think `speed_t` is the type for the non-encoded baud rate values (e.g. 38400 or 115200 which require 32 bits) versus the `ospeed` value which is used as the encoded (termios.c_flag & CBAUD) value (e.g. one of the B9600 etc values) and then used as an index in the baud rate divisor table in the kernel.

Thank you!

Reply:
I will just drop the commit with the #include changes.  It's outside the scope of this PR anyway.  It came from a branch I have with misc portability tweaks, so that I can build portions for modern Linux and use valgrind.  I didn't know that about `extern short speed`, thanks.